fe/fe-core/src/test/java/com/starrocks/alter/CompactionHandlerTest.java

```diff
@@ -79,68 +73,44 @@ public static void beforeClass() throws Exception {
         List<AlterClause> alterList = Collections.singletonList(compactionClause);
         try {
-            new Expectations() {
-                {
-                    compactionMgr.triggerManualCompaction((PartitionIdentifier) any);
-                    times = 1;
-                }
-            };
             compactionHandler.process(alterList, db, olapTable);
+            Assert.assertEquals(expectedValue, GlobalStateMgr.getCurrentState().getCompactionMgr().getPartitionStatsCount());
         } catch (UserException e) {
             e.printStackTrace();
             Assert.fail("process should not throw exceptions here");
         }
     }
 
     @Test
-    public void testProcessCompactionClauseWithTwoPartitionForLakeTable() {
-        List<String> partitionNames = Arrays.asList("p1", "p2");
-        CompactionClause compactionClause = new CompactionClause(partitionNames, true, null);
-        List<AlterClause> alterList = Collections.singletonList(compactionClause);
-        try {
-            new Expectations() {
-                {
-                    compactionMgr.triggerManualCompaction((PartitionIdentifier) any);
-                    times = 2;
-                }
-            };
-            compactionHandler.process(alterList, db, olapTable);
-        } catch (UserException e) {
-            e.printStackTrace();
-            Assert.fail("process should not throw exceptions here");
-        }
+    public void testProcessCompactionClauseWithOnePartitionForLakeTable() {
+        processAlterClauses(List.of("p1"), 1);
     }
 
+    @Test
+    public void testProcessCompactionClauseWithTwoPartitionForLakeTable() {
+        processAlterClauses(Arrays.asList("p1", "p2"), 2);
+    }
 
     @Test
     public void testProcessCompactionClauseWithNoPartitionsForLakeTable() {
-        List<String> partitionNames = Collections.emptyList();
-        CompactionClause compactionClause = new CompactionClause(partitionNames, true, null);
-        List<AlterClause> alterList = Collections.singletonList(compactionClause);
-        try {
-            new Expectations() {
-                {
-                    compactionMgr.triggerManualCompaction((PartitionIdentifier) any);
-                    minTimes = 2;
-                }
-            };
-            compactionHandler.process(alterList, db, olapTable);
-        } catch (UserException e) {
-            e.printStackTrace();
-            Assert.fail("process should not throw exceptions here");
-        }
+        // compaction should trigger on all partitions
+        processAlterClauses(Collections.emptyList(), 2);
     }
 
     @Test(expected = IllegalStateException.class)
     public void testProcessNonCompactionClause() {
         AlterClause nonCompactionClause = mock(AlterClause.class);
-        List<AlterClause> alterList = Collections.singletonList(nonCompactionClause);
+        List<AlterClause> alterList = List.of(nonCompactionClause);
```

Comment:
this is ok for jdk9+, will fail compile on jdk8 (for branch-3.2 and early versions)

Reply:
> this is ok for jdk9+, will fail compile on jdk8 (for branch-3.2 and early versions)

thanks. use collections api instead